test/api/users_test.rb

```diff
@@ -31,11 +31,51 @@ def create_user
   # GET tests
   # ========================================================================
 
+  # Get users' details
   def test_get_users
     get with_auth_token '/api/users'
+    expected_data = User.all
+
+    # Check if the request get through
     assert_equal 200, last_response.status
+
+    # Check if the GET returned the exact number of users
+    assert_equal expected_data.count, last_response_body.count
+
+    # What are the keys we expect in the data that match the model - so we can check these
+    response_keys = %w(first_name last_name email student_id nickname receive_task_notifications receive_portfolio_notifications receive_feedback_notifications opt_in_to_research has_run_first_time_setup)
+
+    # Loop through all of the responses
+    last_response_body.each do | data |
+      # Find the matching user, by id from response
+      user = User.find(data['id'])
+      # Match json with object
+      assert_json_matches_model(data, user, response_keys)
+    end
   end
 
+  # Get a user's details
+  def test_get_a_users_details
+    expected_user = User.second
+
+    # perform the GET 
+    get with_auth_token"/api/users/#{expected_user.id}"
+    returned_user = last_response_body
+
+    # Check if the call succeeds
+    assert_equal 200, last_response.status
+    
+    # Check the returned details match as expected
+    response_keys = %w(first_name last_name email student_id nickname receive_task_notifications receive_portfolio_notifications receive_feedback_notifications opt_in_to_research has_run_first_time_setup)
+    assert_json_matches_model(returned_user, expected_user, response_keys)
+  end
+  
+  # Get CSV of all users
+  def test_get_csv
+    get with_auth_token '/api/csv/users'
+    assert_equal 200, last_response.status
```

Comment:
Expand this to check that the CSV contained the right data - use CSV.parse to read the CSV then you can loop over the users etc.

Reply:
I cannot see how this is resolved... Have a look at `check_task_completion_csv`, `test_export_users` and `test_export_task_definitions_csv`

Reply:
There is a whole file of csv tests that need to be updated to check the response returned is appropriate.

Reply:
With this test, I will be spending more time learning from csv tests and will recommit once I could be able to resolve